src/middlewared/middlewared/plugins/enclosure_/nvme.py

```diff
@@ -75,10 +74,10 @@ def map_plx_nvme_impl(self, prod):
                 pass
             else:
                 for child in physical_node.children:
-                    if not self.RE_NVME.fullmatch(child.sys_name):
+                    if child.properties.get("SUBSYSTEM") != "block":
                         continue
 
-                    if (slot := addresses_to_slots.get(child.parent.sys_name.split(".")[0])) is None:
+                    if (slot := addresses_to_slots.get(child.parent.parent.sys_name.split(".")[0])) is None:
```

Comment:
Should we do a `hasattr(` check to make sure these attributes exist? Otherwise, it'll crash with `AttributeError`?